app/src/main/java/com/example/myphonebook/MyRecyclerViewAdapter.java

```diff
@@ -0,0 +1,78 @@
+package com.example.myphonebook;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.recyclerview.widget.RecyclerView;
+
+public class MyRecyclerViewAdapter extends RecyclerView.Adapter<MyRecyclerViewAdapter.ViewHolder> implements ContactListReceiver{
+    private ContactListProvider mData;
+    private LayoutInflater mInflater;
+    private ItemClickListener mClickListener;
+
+    MyRecyclerViewAdapter(Context context) {
+        this.mInflater = LayoutInflater.from(context);
+        this.mData = null;
+    }
+    public void setData(ContactListProvider data){
+        mData=data;
+        this.notifyDataSetChanged();
+    }
+    // inflates the row layout from xml when needed
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = mInflater.inflate(R.layout.contact_entry_layout, parent, false);
+        return new ViewHolder(view);
+    }
+
+    // binds the data to the TextView in each row
+    @Override
+    public void onBindViewHolder(ViewHolder holder, int position) {
+        if(mData!=null) {
+            String name = mData.getContacts().get(position).name();
+            holder.Name.setText(name);
+        }
+    }
+
+    // total number of rows
+    @Override
+    public int getItemCount() {
+        if(mData==null) return 0;
+        return mData.getContacts().size();
+    }
+
+
+    public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        TextView Name;
+
+        ViewHolder(View itemView) {
+            super(itemView);
+            Name = itemView.findViewById(R.id.Name);
+            itemView.setOnClickListener(this);
+        }
+
+        @Override
+        public void onClick(View view) {
+            if (mClickListener != null) mClickListener.onItemClick(view, getAdapterPosition());
+        }
+    }
+
+    // convenience method for getting data at click position
+    Entry getItem(int id) {
+        if(mData==null) return null;
+        return mData.getContacts().get(id);
+    }
+
+    // allows clicks events to be caught
+    void setClickListener(ItemClickListener itemClickListener) {
+        this.mClickListener = itemClickListener;
+    }
+
+    // parent activity will implement this method to respond to click events
+    public interface ItemClickListener {
+        void onItemClick(View view, int position);
+    }
```

Comment:
no need. you can define this behaviour in `onBindViewHolder `. Every `View` has `setOnClickListener`.